src/lib/eslint/CustomEslintEngine.ts

```diff
@@ -0,0 +1,120 @@
+import { Catalog, RuleGroup, Rule, RuleTarget, RuleResult, RuleViolation } from '../../types';
+import {RuleEngine} from '../services/RuleEngine';
+import {CUSTOM_CONFIG, ENGINE} from '../../Constants';
+import {EslintProcessHelper, StaticDependencies} from './EslintProcessHelper';
+import { FileHandler } from '../util/FileHandler';
+import {Logger, SfdxError} from '@salesforce/core';
+
+
+export class CustomEslintEngine implements RuleEngine {
+	private dependencies: StaticDependencies;
+	private helper: EslintProcessHelper;
+	protected logger: Logger;
+
+	getEngine(): ENGINE {
+		return ENGINE.ESLINT_CUSTOM;
+	}
+
+	getName(): string {
+		return this.getEngine().valueOf();
+	}
+
+	isCustomConfigBased(): boolean {
+		return true;
+	}
+
+	async init(): Promise<void> {
+		this.logger = await Logger.child(`eslint-custom`);
+		this.dependencies = new StaticDependencies();
+		this.helper = new EslintProcessHelper();
+	}
+
+	async getTargetPatterns(): Promise<string[]> {
+		return ["**/*.js"]; // TODO: We need a different way to set target pattern. Somehow eslintrc's ignore pattern doesn't work as expected
```

Comment:
I'm looking for ideas here - ideally, I was hoping targetPattern will be picked up from eslintrc.json and we can allow `**/*`. However, I couldn't get it to work as I expected. Should we be getting targetPatterns as another engineOption? Or should we leave it to the users to handle the long list of parse exceptions when eslint picks up files of types it does not understand?

Reply:
What was ESLint doing when you tried to do `**/*`?

Reply:
Eslint would attempt to scan all the files and would throw parse exceptions on file types it doesn't understand (such as `*.xml`and `*.cls`).
Maybe we should run it all and filter out parse exception errors in the results array.